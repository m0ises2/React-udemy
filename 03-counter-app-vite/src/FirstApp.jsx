import PropTypes from "prop-types";

export const FirstApp = ( { title, subtitle, age } ) => {
    return (
        <>
            <h1 data-testid='test title'> { title } </h1>
            <p data-testid='test subtitle'> { subtitle } </p>
            <p> { age } years old </p>
        </>
    ) // Esto es un fragmento
}

FirstApp.propTypes = {
    age: PropTypes.number.isRequired,
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    age: 35,
    subtitle: 'Subtitulo por defecto'
};
