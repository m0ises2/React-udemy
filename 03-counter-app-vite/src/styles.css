html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}