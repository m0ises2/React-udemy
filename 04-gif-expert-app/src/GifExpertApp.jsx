import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);
  
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    
    setCategories([newCategory.toLowerCase(),...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      
      {/* Lista de Gif */}
      { 
        categories.map(category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
        {/* Gif Item */}

    </>
  );
};
