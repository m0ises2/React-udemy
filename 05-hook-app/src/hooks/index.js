export * from './UseCounter';
export * from './useFetch';
export * from './useForm';