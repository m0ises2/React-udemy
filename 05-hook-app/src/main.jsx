import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HooksApp } from "./HooksApp";
import './index.css';
// import { CounterApp } from './01-useState/CounterApp';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { SimpleForm } from './02-useEffect/SimpleForm';
import { SimpleFormWithCustomHook } from './02-useEffect/SimpleFormWithCustomHook';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>

    <SimpleFormWithCustomHook />

  </React.StrictMode>
);
