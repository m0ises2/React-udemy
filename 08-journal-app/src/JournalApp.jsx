import { RouterProvider } from 'react-router-dom';

import { AppRouter } from './router/AppRouter';

export const JournalApp = () => {
  return (<RouterProvider router={AppRouter} />);
}
