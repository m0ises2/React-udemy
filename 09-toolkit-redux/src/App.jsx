import { useSelector, useDispatch } from 'react-redux'
import { increment, incrementBy, decrement} from './store/slices/counter/counterSlice'

import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

function App() {
  const { counter } = useSelector((state) => state.counter)
  const dispatch = useDispatch()

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <p>count is { counter }</p>
      <div className="card">
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>
        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>
        <button onClick={() => dispatch(incrementBy(3))}>
          Increment by 3
        </button>
      </div>
    </>
  )
}

export default App
