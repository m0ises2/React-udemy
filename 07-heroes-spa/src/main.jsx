import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';
import { AppRouter } from './router/AppRouter';
import { AuthProvider } from './auth';

ReactDOM.createRoot(document.getElementById('root')).render(
  <AuthProvider>
    
    <AppRouter />

  </AuthProvider>,
)
