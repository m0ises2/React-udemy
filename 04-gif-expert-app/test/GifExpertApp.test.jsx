import { fireEvent, render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import { GifExpertApp } from "../src/GifExpertApp";

describe('pruebas de GifExpertApp component', () => {
    const newCategory = 'One Punch';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);

        expect(container).toMatchSnapshot();
    });

    test('debe existir el titulo GitExpertApp', () => {
       const title = 'GitExpertApp';

        render(<GifExpertApp />);
        expect(screen.getByRole('heading', { level: 1 } ).innerHTML).toBe(title);
    });

    test('debe mostrar la categoria ingresada en el DOM', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);

        const categoryHeading = screen.getByRole('heading', { level: 3 });

        expect(categoryHeading.innerHTML).toContain(`${newCategory.toLowerCase()}`);
    });

    test('debe excluir la categoria si ya existe', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);

        const categoryHeading = screen.getAllByRole('heading', { level: 3 });

        expect(categoryHeading.length).toBe(1);
    });

    test('debe cargar las imagenes de la categoria ingresada una vez que el loading desaparece', async () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: newCategory } } );
        fireEvent.submit(form);

        await waitForElementToBeRemoved(screen.getByRole('heading', { level: 2 }));

        const allImgs = screen.getAllByRole('img');

        expect(allImgs.length).toBe(5);
    });
});
