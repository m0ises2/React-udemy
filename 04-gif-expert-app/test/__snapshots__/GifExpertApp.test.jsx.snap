// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de GifExpertApp component debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
